src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -121,9 +125,12 @@ protected void Dispose(bool disposing)
                     if (_readAheadTask != null)
                     {
                         ValueTask<int> t = _readAheadTask.GetValueOrDefault();
-                        if (t.IsCompleted && !t.IsCompletedSuccessfully)
```

Comment:
>if (t.IsCompleted && !t.IsCompletedSuccessfully) [](start = 24, length = 48)

Q.: why separate "if" statements instead of the logical "and"?

Reply:
This is actually fixing a bug.  We don't want to execute the else at all if it's already completed; the way it was written, we were executing the else if it was already completed successfully.